readthedocs/projects/forms.py

```diff
@@ -541,6 +551,72 @@ def __init__(self, *args, **kwargs):
         except AddonsConfig.DoesNotExist:
             self.fields["enabled"].initial = False
 
+        fieldsets = [
+            Fieldset(
+                _("Global"),
+                "enabled",
+            ),
+            Fieldset(
+                _("Analytics"),
+                HTML(
+                    "<p>Analytics addon allows you to store traffic analytics in your project.</p>"
+                ),
+                *[field for field in self.fields if field.startswith("analytics_")],
+            ),
+            Fieldset(
+                _("DocDiff"),
+                HTML(
+                    "<p>DocDicc addon helps you to review changes from PR by visually showing the differences.</p>"
+                ),
+                *[field for field in self.fields if field.startswith("doc_diff_")],
+            ),
+            Fieldset(
+                _("Flyout"),
+                HTML(
+                    "<p>Flyout addon shows a small menu at bottom-right of each page with useful links to switch between versions and translations.</p>"
+                ),
+                *[field for field in self.fields if field.startswith("flyout_")],
+            ),
+            Fieldset(
+                _("Hotkeys"),
+                HTML(
+                    "<p>Hotkeys addon enables keyboard shortcuts to access other addons quickly.</p>"
+                ),
+                *[field for field in self.fields if field.startswith("hotkeys_")],
+            ),
+            Fieldset(
+                _("Search"),
+                HTML(
+                    "<p>Search addon improves your search experience with the search-as-you-type feature.</p>"
+                ),
+                *[field for field in self.fields if field.startswith("search_")],
+            ),
+            Fieldset(
+                _("Warning on PR previews"),
+                HTML("<p>Shows a warning notification on PR previews.</p>"),
+                *[
+                    field
+                    for field in self.fields
+                    if field.startswith("external_version_warning_")
+                ],
+            ),
+            Fieldset(
+                _("Warning on non-stable/latest versions"),
+                HTML(
+                    "<p>Shows a warning notification stable and latest versions letting the user know they may be reading an outdated page or a non-released one.</p>"
+                ),
+                *[
+                    field
+                    for field in self.fields
+                    if field.startswith("stable_latest_version_warning_")
+                ],
+            ),
+        ]
```

Comment:
Noted at https://github.com/readthedocs/ext-theme/pull/262#issuecomment-1894338748

Two things that are important here:

First, I think this interface will eventually use a tabbed interface, not fieldsets, or might even use separate form/views entirely -- we'll know more once we start adding deeper configuration options here. But for that, I am gathering fieldsets will probably be removed in the future.

Second, I wanted to discuss avoiding (and probably removing) use of Crispy layouts and the layout API. Having HTML authoring removed from template code is a bit awkward already, just giving the code focus and repository split, but it gets much worse if we want JS/Knockout/etc code in the form.

For that, I might suggest keeping this incredibly simple for a first pass and removing fieldsets and Python HTML authoring.

Reply:
Yeah, that seems like plenty :+1: 

It will look minimal, but should be fine -- especially with a fix for block display checkboxes. We definitely will want to improve on that later in some way in the future, and we can start mocking out what that looks like once we know what other configuration fields we want there.

Reply:
I'm not sure how to fix the inline checkbox CSS issue, but I found _where_ is the issue at least.

```css
.ui.checkbox {
  display: inline-block;
}
```

Removing that property it displays as we want, I'd say.

![Screenshot_2024-01-18_13-20-22](https://github.com/readthedocs/readthedocs.org/assets/244656/3ee73a69-f644-4592-b867-9dbd68617832)

Reply:
Yeah, that sounds good. I agree it's probably minor and I should be able to squeeze a fix into the crispy field template before deploy